libraries/c_sdk/aws/defender/src/aws_iot_defender_mqtt.c

```diff
@@ -37,30 +35,35 @@
 
 #define TOPIC_SUFFIX_REJECTED    TOPIC_SUFFIX_PUBLISH "/rejected"
 
-extern AwsIotDefenderStartInfo_t _startInfo;
+/*-----------------------------------------------------------*/
 
-/* defender internally manages network and mqtt connection */
-static IotMqttConnection_t _mqttConnection = IOT_MQTT_CONNECTION_INITIALIZER;
+extern IotMqttCallbackInfo_t _acceptCallbackInfo;
+extern IotMqttCallbackInfo_t _rejectCallbackInfo;
+extern AwsIotDefenderStartInfo_t _startInfo;
 
 static char * _pPublishTopic = NULL;
+static size_t _publishTopicLength = 0;
 
 static char * _pAcceptTopic = NULL;
+static size_t _acceptTopicLength = 0;
 
 static char * _pRejectTopic = NULL;
+static size_t _rejectTopicLength = 0;
 
 /*-----------------------------------------------------------*/
 
 AwsIotDefenderError_t AwsIotDefenderInternal_BuildTopicsNames( void )
 {
     AwsIotDefenderError_t returnedError = AWS_IOT_DEFENDER_SUCCESS;
 
-    const char * pThingName = _startInfo.mqttConnectionInfo.pClientIdentifier;
-    uint16_t thingNameLength = _startInfo.mqttConnectionInfo.clientIdentifierLength;
+    const char * pThingName = _startInfo.pClientIdentifier;
+    uint16_t thingNameLength = _startInfo.clientIdentifierLength;
+    size_t topicPrefixLength = strlen( TOPIC_PREFIX );
 
     /* Calculate topics lengths. Plus one for string terminator. */
-    size_t publishTopicLength = strlen( TOPIC_PREFIX ) + thingNameLength + strlen( TOPIC_SUFFIX_PUBLISH ) + 1;
-    size_t acceptTopicLength = strlen( TOPIC_PREFIX ) + thingNameLength + strlen( TOPIC_SUFFIX_ACCEPTED ) + 1;
-    size_t rejectTopicLength = strlen( TOPIC_PREFIX ) + thingNameLength + strlen( TOPIC_SUFFIX_REJECTED ) + 1;
+    size_t publishTopicLength = topicPrefixLength + thingNameLength + strlen( TOPIC_SUFFIX_PUBLISH ) + 1;
```

Comment:
Minor. For ```strlen``` usage in general. The length of the macro defined string is known at compile time, and these strings are likely not going to be changed (since that'll require server side change as well). Given that, I think ```strlen``` is unnecessary, and we could further remove dependency on ```string.h```. No harm is done, just commenting as a minimalist.